### ISSUE TITLE
setContent / bindPopup incompatible with templates?

### ISSUE THREAD

**Issue by RobertLowe** [bevanhunt/meteor-leaflet#1]
Hi,

Any ideas on how to render a meteor template into a popup (I also want events bound on the template to work)?

I'm unable to open a popup with a dom fragment `frag = Meteor.render(Template.details(an_event)).

Cheers,
- Rob

**Comment by vchigne** [comment]
Actually, it would be great to have reactive content in a leaflet popup. Maybe a way to send a template name and parameters to the html property of the popup maintaning reactivity. I tried to do it with a iframe pointing to a meteor route but it still looses reactivity.

**Comment by RobertLowe** [comment]
This is a very old issue, I think I can be worked around with:

    <body>
      {{> map}}

      <div style='display: none'>
        {{> thing}}
      </div>
    </body>

    <template name='thing'>
      <div id='thing'>
        <p> hello world </p>
      </div>
    </template>

    Template.thing.created = function () {
      var popup = L.popup().setLatLng(<somewhere>)

      // change happens
      this.autorun(function () {
        popup.setContent($('#thing')[0])
      });
    }

I'm not sure if events will work as expected. `autorun` is very useful.

**Comment by vchigne** [comment]
I'll try to apply that concept. It's kind of hard because irontouter apps don't have <body>, so I have to find where to put the div.

**Comment by RobertLowe** [comment]
Okay, I did some digging, this is fine in current Meteor, it's a pick tricky though:

Say you have a marker on the map:

      var marker = L.marker({lat: 49.26814058915736, lng: -123.12807083129883}).addTo(map)

You can then bind the marker with a popup like so via the `Blaze.renderWithData` API:

      containerNode = document.createElement('div'); // wrapping node for bindPopup;
      // Which template to use for the popup? Some data for it, and attach it to the containerNode
      Blaze.renderWithData(Template.popup, dataContext, containerNode);
      // Finally bind the containerNode to the popup
      marker.bindPopup(containerNode).openPopup()

**Comment by vchigne** [comment]
Works great! Maybe it could be added as a new function, like L.reactivePopup('templateName').
Thanks

**Comment by RobertLowe** [comment]
@bevanhunt ping any thoughts?

have fun @vchigne :package: :+1:

**Comment by soccertwin03** [comment]
@RobertLowe I followed the above and I get the popup to show the template and data from the collection.  However, when I modify the data in the collection, it doesn't update the data in the popup. 

My workflow is as follows... user clicks on a marker and gets the pop-up, which contains the name field from the collection.  When the user clicks on the name field, they can modify the name, which triggers an update on the collection.  Once the update occurs, I am able to verify that the collection got updated via the console but the name field doesn't get updated until a browser refresh.  

Any thoughts?
